### PR TITLE
Implement `asarray`

### LINKED ISSUE #308: Implement `asarray`

### PR DESCRIPTION
**Pull request by bveeramani**
Closes #308

### CODE REVIEW

File: nums/numpy/api/creation.py [hunk 1]
```diff
@@ -1116,3 +1116,77 @@ def zeros_like(prototype, dtype=None, order="K", shape=None):
...
+    asfortranarray : Convert input to an ndarray with column-major
+                     memory order.
+    asarray_chkfinite : Similar function which checks input for NaNs and Infs.
+    fromiter : Create an array from an iterator.
+    fromfunction : Construct an array by executing a function on grid
+                   positions.
+
+    Examples
+    --------
+    Convert a list into an array:
```
Comment (briancpark) [inline nums/numpy/api/creation.py:1156]: Add

```
The doctests shown below are copied from NumPy.
They won’t show the correct result until you operate ``get()``.
```

File: nums/numpy/api/creation.py [hunk 2]
```diff
@@ -1116,3 +1116,77 @@ def zeros_like(prototype, dtype=None, order="K", shape=None):
     if order is not None and order != "K":
         raise NotImplementedError("Only order='K' is supported.")
     return zeros(shape, dtype)
+
+
+def asarray(a, dtype=None, order=None, *, like=None):
+    """Convert the input to an array.
+
```
Comment (briancpark) [inline nums/numpy/api/creation.py:1123]: Add the lines:

```
This docstring was copied from numpy.asarray.

Some inconsistencies with the NumS version may exist.
```

This is just to be consistent with the other documentation.

File: nums/numpy/api/creation.py [hunk 3]
```diff
@@ -1116,3 +1116,77 @@ def zeros_like(prototype, dtype=None, order="K", shape=None):
...
+        subclass of ndarray, a base class ndarray is returned.
+
+    Raises
+    ------
+    NotImplementedError
+        If you pass a non-`None` value to `order` or `like`.
+
+    See Also
+    --------
+    asanyarray : Similar function which passes through subclasses.
```
Comment (briancpark) [inline nums/numpy/api/creation.py:1147]: Delete other APIs we don't officially support.

**Review by briancpark — CHANGES_REQUESTED** [review]
Requested some changes related to documentation.

File: nums/numpy/api/creation.py (hunk 2 above)
Comment (bveeramani) [reply nums/numpy/api/creation.py:1123]: I manually checked the docstring. There shouldn't be an inconsistencies

File: nums/numpy/api/creation.py (hunk 3 above)
Comment (bveeramani) [reply nums/numpy/api/creation.py:1147]: I deleted the two that we'll never support. The others we'll conceivably implement in the future. Do you think I should remove them now and add them back later, or just keep them?

File: nums/numpy/api/creation.py (hunk 1 above)
Comment (bveeramani) [reply nums/numpy/api/creation.py:1156]: I manually ran all the doctests. They all work correctly without changes

File: nums/numpy/api/creation.py (hunk 3 above)
Comment (bveeramani) [reply nums/numpy/api/creation.py:1147]: I figure if we delete them now we'll probably forget to add it back later

Comment (briancpark) [reply nums/numpy/api/creation.py:1147]: If you want to implement them, then sure leave them behind. But I believe they'll link to NumPy docs, so make sure to recompile locally to make sure

File: nums/numpy/api/creation.py (hunk 1 above)
Comment (briancpark) [reply nums/numpy/api/creation.py:1156]: This is just a warning on all of the NumPy API docstrings

File: nums/numpy/api/creation.py (hunk 2 above)
Comment (briancpark) [reply nums/numpy/api/creation.py:1123]: This is just a warning on all of the NumPy API docstrings

File: nums/numpy/api/creation.py [hunk 4]
```diff
@@ -1116,3 +1116,74 @@ def zeros_like(prototype, dtype=None, order="K", shape=None):
...
+    a : array_like
+        Input data, in any form that can be converted to an array.  This
+        includes lists, lists of tuples, tuples, tuples of tuples, tuples
+        of lists and ndarrays.
+    dtype : data-type, optional
+        By default, the data-type is inferred from the input data.
+
+    Returns
+    -------
+    out : ndarray
```
Comment (briancpark) [inline nums/numpy/api/creation.py:1135]: Change this to `BlockArray` and any other mentions of `ndarray` to `BlockArray`

File: nums/numpy/api/creation.py (hunk 1 above)
Comment (bveeramani) [reply nums/numpy/api/creation.py:1156]: You don't need to call `get` for two of the three examples, so I'm not sure if we should include it.

```
>>> a = nps.array([1, 2])
>>> nps.asarray(a) is a
True
```

```
 >>> a = nps.array([1, 2], dtype=np.float32)
 >>> nps.asarray(a, dtype=np.float32) is a
True
>>> nps.asarray(a, dtype=np.float64) is a
False
```

File: nums/numpy/api/creation.py (hunk 3 above)
Comment (bveeramani) [reply nums/numpy/api/creation.py:1147]: Okay, sounds good. I'll compile the documentation and check

File: nums/numpy/api/creation.py (hunk 4 above)
Comment (bveeramani) [reply nums/numpy/api/creation.py:1135]: Yeah. Was talking to @elibol about potentially renaming `BlockArray` to `ndarray`, but you're right. Shouldn't prematurely change the name.

Comment (bveeramani) [reply nums/numpy/api/creation.py]: Fixed

File: nums/numpy/api/creation.py (hunk 2 above)
Comment (bveeramani) [reply nums/numpy/api/creation.py:1123]: I hear you. I see how adding the lines improves consistency. That being said, I don't think we should warn users about inconsistencies if we know there are none.

Comment (briancpark) [reply nums/numpy/api/creation.py:1123]: Sure, if @elibol approves, then I'm fine with not including this and resolving the followup with the doctest warning.

Other than that looks consistent with the documentation

Comment (elibol) [reply nums/numpy/api/creation.py:1123]: sounds good to me.

**Review by elibol — APPROVED** [review]
LGTM.

**Pull request merged**